src/status_im/contexts/wallet/wallet_connect/events/core.cljs

```diff
@@ -58,6 +58,23 @@
              constants/wallet-connect-session-delete-event
              #(rf/dispatch [:wallet-connect/on-session-delete %])]]]})))
 
+(rf/reg-event-fx
+ :wallet-connect/unregister-event-listeners
+ (fn [{:keys [db]}]
+   (let [web3-wallet (get db :wallet-connect/web3-wallet)]
+     {:fx [[:effects.wallet-connect/unregister-event-listener
+            [web3-wallet
+             constants/wallet-connect-session-proposal-event
+             #(rf/dispatch [:wallet-connect/on-session-proposal %])]]
+           [:effects.wallet-connect/unregister-event-listener
+            [web3-wallet
+             constants/wallet-connect-session-request-event
+             #(rf/dispatch [:wallet-connect/on-session-request %])]]
+           [:effects.wallet-connect/unregister-event-listener
+            [web3-wallet
+             constants/wallet-connect-session-delete-event
+             #(rf/dispatch [:wallet-connect/on-session-delete %])]]]})))
+
```

Comment:
not sure if this right, since you're passing a different handler when un-registering the event, and usually js event emitters unregister handlers by handler reference. Here the content of the handlers (when registering and unregistering) is the same, but the function references are different. You can extract the handlers and pass them to the `(register/unregister)-event-listeners` fx.

Reply:
@mohsen-ghafouri could please ping me up once review comments are resolved. After that I will continue testing. Thanks.

Reply:
Done @pavloburykh @clauxx